test.js

```diff
@@ -1,212 +1,242 @@
+describe("collect", function () {
     var result = {};
     var source = {
-        apps: [ "myApp1", "myApp2" ],
+        apps: ["myApp1", "myApp2"],
     }
 
     var response = {
-        "myApp1": { data: dataGenerator( 100, null ), name: "myApp1" },
-        "myApp2": { data: dataGenerator( 100, null ), name: "myApp2" }
+        "myApp1": { data: dataGenerator(100, null), name: "myApp1" },
+        "myApp2": { data: dataGenerator(100, null), name: "myApp2" }
     }
 
-    before( initialize( result, response, source ) )
-    after( reset )
+    before(initialize(result, response, source))
+    after(reset)
 
-    it( "should read 200 rows for two apps", function() {
-        assert.equal( result.data.length, 200 );
-        assert.equal( result.data[ 0 ].appName, "myApp2" );
-        assert.equal( result.data[ 100 ].appName, "myApp1" );
+    it("should read 200 rows for two apps", function () {
+        assert.equal(result.data.length, 200);
+        assert.equal(result.data[0].appName, "myApp2");
+        assert.equal(result.data[100].appName, "myApp1");
     })
 })
 
-describe( "Fetch beginning of time", function () {
+describe("Fetch beginning of time", function () {
     var result = {};
     var source = {
-        apps: [ 'myApp' ],
+        apps: ['myApp'],
         ignoreMissing: true
     }
 
-    var response = { "myApp": { error: {}, name: "myApp", data: dataGenerator( 1, null ) } }
+    var response = { "myApp": { error: {}, name: "myApp", data: dataGenerator(1, null) } }
 
-    before( initialize( result, response, source, true ) )
-    after( reset )
+    before(initialize(result, response, source, true))
+    after(reset)
 
-    it( 'Fetch insights from beginning of time', function () {
+    it('Fetch insights from beginning of time', function () {
         const parts = calledUrl.split('?')
         const parsed = queryString.parse(parts[1])
         assert.equal(Boolean(parsed.since), false)
     })
 })
 
-describe( "Fetch x Days ago", function () {
+describe("Fetch x Days ago", function () {
     var result = {};
     var source = {
-        apps: [ 'myApp' ],
+        apps: ['myApp'],
         ignoreMissing: true
     }
 
-    var response = { "myApp": { error: {}, name: "myApp", data: dataGenerator( 1, null ) } }
+    var response = { "myApp": { error: {}, name: "myApp", data: dataGenerator(1, null) } }
 
-    before( initialize( result, response, source ) )
-    after( reset )
+    before(initialize(result, response, source))
+    after(reset)
 
-    it( 'Fetch insights for past x days', function () {
+    it('Fetch insights for past x days', function () {
         const parts = calledUrl.split('?')
         const parsed = queryString.parse(parts[1])
         assert.equal(Boolean(parsed.since), true)
     })
 })
 
-describe( 'Multiple access tokens', function () {
+describe("Fetch date range", function () {
+    // Test date range between today and 30 days ago
+    var since = new Date();
+    var until = Date.now();
```

Comment:
Above we are doing `var since = new Date(options.since)` so `options.since` should be a string and not a date. The same goes for `since`
```
const since = "2018-08-20"
const until = "2018-09-20"
```